include/claws/array_ops.hpp

```diff
@@ -170,5 +170,20 @@ namespace claws
     SCALAR_ARRAY_OPERATOR_DEF(>>);
   }
 
+  namespace impl
+  {
+    template<class Func, class T, std::size_t count, std::size_t... indices>
+    std::array<T, count> init_array(Func functor, std::index_sequence<indices...>)
+    {
+      return std::array<T, count>{(indices, functor())...};
+    }
+  }
+
+  /// \brief initialises each element of the array with the given functor
+  template<class Func, class T, std::size_t count>
+  std::array<T, count> init_array(Func functor)
+  {
+    return impl::init_array(functor, std::make_index_sequence<count>{});
```

Comment:
I'm suprised too. I'm pretty sure it compiled because I remember using it in the compositor. I'll have to take a closer look at it as soon I have my PC.

Reply:
Ok, seems I was wrong. I'll close this PR for the moment.